packaging/ansible-runner-service-project/project/roles/gluster-replace-peers/tasks/peers.yml

```diff
@@ -106,60 +103,61 @@
 
     - name: Check whether peer file of cluster_node is available
       connection: ssh
-      stat:
+      ansible.builtin.stat:
         path: "{{ glusterd_libdir }}/peers/{{ current_uuid }}"
       register: peerfileres
       delegate_to: "{{ gluster_cluster_node_2 | mandatory }}"
 
     - name: Fetch the peer file of the current node
       connection: ssh
-      fetch:
+      ansible.builtin.fetch:
         src: "{{ glusterd_libdir }}/peers/{{ current_uuid }}"
         dest: "{{ peer_tmp_dir }}/"
-        flat: yes
+        flat: true
       delegate_to: "{{ gluster_cluster_node_2 | mandatory }}"
       when: peerfileres.stat.isreg is defined
 
     - name: Fetch the peer file of the current node
       connection: ssh
-      fetch:
+      ansible.builtin.fetch:
         src: "{{ glusterd_libdir }}/peers/{{ current_uuid }}"
         dest: "{{ peer_tmp_dir }}/"
-        flat: yes
+        flat: true
       delegate_to: "{{ gluster_cluster_node | mandatory }}"
       when: peerfileres.stat.isreg is not defined
 
     - name: Remove the old node uuid from the extracted peer details
       connection: ssh
-      file:
+      ansible.builtin.file:
         path: "{{ peer_tmp_dir }}/{{ old_uuid }}"
         state: absent
       delegate_to: "{{ gluster_cluster_node }}"
 
     - name: Copy all the peer files to the new host
       connection: ssh
-      copy:
+      ansible.builtin.copy:
         src: "{{ peer_tmp_dir }}/"
         dest: "{{ glusterd_libdir }}/peers/"
+        mode: preserve
```

Comment:
@gobindadas What should be the actual permissions? Copying permissions from local temporary directory seems dangerous to me

Reply:
I think the safer way is to do 'mode' with required permissions

Reply:
fixed